### ISSUE TITLE
No route matches

### ISSUE THREAD

**Issue by Brigithubert** [danmunn/redmine_dmsf#625]
I just can't get a current version of redmine to run with dmsf. :(

Now the error is like this:
ActionController::RoutingError (No route matches [GET] "/plugin_assets/redmine_dmsf/javascripts/redmine_dmsf.js")

Is this related to #582?

Will I have to use version 1.5.9?

I'll happily provide more information if necessary and/or helpful.

Redmine 3.3.1
DMSF 1.5.8
Ubuntu 16.04
ruby 2.3

**Comment by picman** [comment]
No it isn't, I think.
Is the file physically in the file system? Is readable for the web server user?

**Comment by Brigithubert** [comment]
You're response time is incredible: it's Christmas! :) Thank you so much for taking the time!

In fact, there is no directory under the redmine root folder called "plugin_assets". The requested file resides under `<redmine-root>/plugins/redmine_dmsf/assets/javascripts/`. I have no idea, though, where to correct this.

**Comment by picman** [comment]
After a rails application starts, assets files are copied to the public folder. Check the public folder when the application is running.

**Comment by Brigithubert** [comment]
Ok, I see. The path `<redmine-root>/public/plugin_assets/`remains empty even after startup of the application. The complete redmine installation directory (i.e. `<redmine-root>`) is owned by the webserver user and thus readable and writable.

I created a symlink to the plugin's assets directory. It seems that now the required files can be found -- I am getting a different error, though:
`ActionController::RoutingError (uninitialized constant DmsfController)` ...

May be it is worth noting, that I am running thin as application server with lighttpd as web server?!

**Comment by picman** [comment]
If assets are not copied to the public directory after the application starts, something is wrong with your application. Redmine itself without the plugin works fine? Other plugins work? Have you followed installation instruction for plugins?

**Comment by Brigithubert** [comment]
It is running fine w/out any plugin. -- Right now, I have only the "wiki extensions plugin" running. That is running fine (assets got copied to the required folder).

Some other plugins do indeed cause similar trouble, e.g. redmine_my_page-master

What else can I do? I got a fresh copy of redmine (didn't use the xenial package), unzipped it, setup thin as app server, lighttpd forwards to the (local) thin servers.

The database was taken from my "old" redmine system on another server. I ran
`bundle exec rake db:migrate RAILS_ENV=production`
`bundle exec rake redmine:plugins:migrate RAILS_ENV=production`,
basically followed the RedmineUpgrade instructions.

Whenever I try to install the plugins (one by one to find out, what is working), I get 404 when trying to configure them.

**Comment by picman** [comment]
I've no idea what could be wrong. I've never tested it on lighttpd yet. Sorry.

**Comment by Brigithubert** [comment]
You may close this thread -- it was a typical "pebkac".

I did not rename the plugin from redmine_dmsf-master to redmine_dmsf... After having done this, it works just right...

Thank you for your incredible patience with me!

Have a happy new year.
